### ISSUE TITLE
Fails on Mobile

### ISSUE THREAD

**Issue by mattruddy** [FilipChalupa/pwa-install-handler#1]
I'm getting a "Unhandled Rejection: Not allowed to prompt" on mobile. Can prompting work on mobile?

**Comment by FilipChalupa** [comment]
You can't prompt while `pwaInstallHandler.canInstall()` returns `false`. This happens on browsers without PWA+prompt support (https://developer.mozilla.org/en-US/docs/Web/API/Window/onbeforeinstallprompt#Browser_compatibility e.g. Firefox, Safari). You must meet browser specific PWA requirements which includes valid webmanifest, service worker.

**Comment by FilipChalupa** [comment]
I highly recommend hiding your custom Add to homescreen button and not calling prompt on page load. Wait for `canInstall === true` callback. 

```
pwaInstallHandler.addListener((canInstall) => {
	$button.style.display = canInstall ? 'inline-block' : 'none'
})
``` 

https://github.com/FilipChalupa/pwa-install-handler#usage

**Comment by mattruddy** [comment]
Thanks, is there a way to tell if the pwa is already installed?

**Comment by FilipChalupa** [comment]
Installed PWA will always report `false` in the listener and `pwaInstallHandler.canInstall()`.

**Comment by mattruddy** [comment]
yes, but then how can you know the difference between if the browers doesnt support prompt() vs its already installed?

**Comment by FilipChalupa** [comment]
To check your app is installed, you can do this:

```javascript
if (window.matchMedia('(display-mode: standalone)').matches) {
  console.log('display-mode is standalone');
}
```
